tools/snp.sh

```diff
@@ -1081,42 +1062,134 @@ attest_guest() {
+   
+    # Update and install necessary packages
+    echo 'Updating package list...'
+    sudo apt-get update
+
+    echo 'Installing necessary packages...'
+    sudo apt-get install -y git build-essential libtss2-dev tpm2-tools
+
+    echo 'Installing Rust...'
+    source ""${HOME}"/.cargo/env" 2>/dev/null || true
+
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -sSf | sh -s -- -y
+    source ""${HOME}"/.cargo/env" 2>/dev/null
+    "    
+
+  # Check if 'snpguest' directory exists and remove it if it does
+  ssh_guest_command "
+    if [ -d \"snpguest\" ]; then
+        echo 'Directory snpguest exists. Removing it...'
+        rm -rf snpguest
+    else
+        echo 'Directory snpguest does not exist.'
+    fi
+"
+
+  # Clone the repository and build the project
+  ssh_guest_command "git clone https://github.com/virtee/snpguest.git"
+  ssh_guest_command "cd snpguest  &&  source \$HOME/.cargo/env &&  cargo build --release"
+}
 
-  # Use sev-snp-measure utility to calculate the expected measurement
-  local expected_measurement=$(generate_snp_expected_measurement)
-  echo -e "\nExpected Measurement (sev-snp-measure):  ${expected_measurement}"
+Verify_measurement(){
+
+  # Define the working directory and the output file
+  OUTPUT_FILE="$SETUP_WORKING_DIR/make_output.txt"
+  # Define the firmware file path
+  FW_FILE="$SETUP_WORKING_DIR/AMDSEV/ovmf/Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd"
+
+  cd "$SETUP_WORKING_DIR"/AMDSEV/svsm
+
+  # Run the make command and save the output to a file
+  FW_FILE="$FW_FILE" make &> "$OUTPUT_FILE"
+
+  # Initialize the variable for Launch Digest
+  launch_digest=""
+
+  # Read the output file line by line to find the Launch Digest
+  while IFS= read -r line; do
+    if [[ "$line" == *"Launch Digest:"* ]]; then
+      launch_digest=$(echo "$line" | awk '{print $3}')
+      break
+    fi
+  done < "$OUTPUT_FILE"
+
+  # Output the Launch Digest to verify
+  echo "Launch Digest: $launch_digest"
+
+  # Use igvmmeasure utility to calculate the expected measurement
+  local expected_measurement=$launch_digest
+  echo -e "\nExpected Measurement (igvmmeasure):  ${expected_measurement}"
 
   # Parse the measurement out of the snp report
-  local snpguest_report_measurement=$(ssh_guest_command \
-    "./snpguest display report attestation-report.bin \
-    | tr '\n' ' ' \
-    | sed \"s|.*Measurement:\(.*\)Host Data.*|\1\n|g\" \
-    | sed \"s| ||g\"")
-
-  # Remove any special characters and print the value
-  snpguest_report_measurement=$(echo ${snpguest_report_measurement} | sed $'s/[^[:print:]\t]//g')
+  scp_guest_command "${GUEST_USER}@localhost:$HOME/attestation-report.bin" "$SETUP_WORKING_DIR/AMDSEV/svsm"
+
+  cd "$SETUP_WORKING_DIR"/AMDSEV/svsm
+  # Assign the output of the hexdump command to a variable
+  snpguest_report_measurement=$(hexdump -s 0x90 -n 48 -v -e '48/1 "%02X"' attestation-report.bin)
+
+  # Print the variable to verify its content
+  echo "$snpguest_report_measurement"
```

Comment:
Do we need to be outputting this value? We are doing it on the subsequent line...